### PR TITLE
Support multiple notes per bibliography resource

### LINKED ISSUE #685: Analysis: Support multiple annotations per bibliography resource
Our BibTeX reader only captures a single annotation! e.g. [book.bib](https://github.com/sul-dlss/exhibits/blob/676-bib-show-page-annotations/spec/fixtures/bibliography/book.bib) has 3 `annote` values. When I checked our our BibTeX `record`, we only get the last annotation.

## Design
show page

design notes

#### BibTeX:
```
@book{http://zotero.org/groups/1051392/items/TXXUJDG2,
	address = {Coimbra},
	title = {A {Carta} ou {Memória} do {Cruzado} {Inglês} {R}. para {Osberto} de {Bawdsey} sobre a conquista de {Lisboa} em 1147},
        …
	annote = { KAW},
	annote = {CCC MS 470 [p. 9, and referred to (without MS number) throughout]reproductions from CCC MS 470, ff. 125-146, in the appendix},
	annote = {The following values have no corresponding Zotero field:custom1: 23}
}
```
#### BibTeX::Entry
```
#<BibTeX::Entry 
  address = Coimbra, 
  title = A Carta ou Memória do Cruzado Inglês R. para Osberto de Bawdsey sobre a conquista de Lisboa em 1147, 
  …
  annote = The following values have no corresponding Zotero field:custom1: 23
>
```

### Possible routes forward (please feel free to edit and add notes below)

1. mess with our BibTex translator so that we get something more like:
`annote = {KAW, The following values have no corresponding Zotero field:custom1: 23}}` and possibly make a PR to [zotero/translators](https://github.com/zotero/translators)) (@camillevilla)
2. adjust our Traject settings to deal with multiple `annote` values (in consultation with @cmh2166) 
3. issue a PR to bibtex-ruby to handle concatenation of multiple values with the same key (suggested by @drh-stanford)
4. [zotero-better-bibtex](https://github.com/retorquere/zotero-better-bibtex) (@mejackreed)

### PR DESCRIPTION
**Pull request by camillevilla**
Closes #685 
Design #605 

This PR:
- updates our Zotero --> BibTeX --> Traject --> Solr pipeline so that all notes are captured
- updates all fixtures with `annote` values (multiple notes are separated by `\textbar` (`|` in LaTeX land)
- adds paragraph separators to the notes (pending discussion below)

## Todo
- [x] [ Zotero --> BibTeX translator](https://github.com/camillevilla/bibtex-translator) that separates notes with `|` character
  - [ ] PR in zotero/translators
  - [x] Add translator to team Google drive
- [x] BibTeX --> Traject
- [x] update fixtures `annote` section
- [x] Blacklight config / helper for presenting notes on separate lines in show page (1: `<ul>` ideal, 2: `<p>` ok #605)
- [x] bib resource integration tests
- [x] unit test

## Questions:

- [x] How do existing exhibits use the `general_notes_ssim` field? Is this too much top-down control over how notes are displayed? 

- [x] How can I refactor this and make it more Blacklight-ish? 

## Before 
Before [6d38407](https://github.com/sul-dlss/exhibits/pull/715/commits/6d384075ab7c3a41968c05c05f8ce629a1d09fe0) we were only captured one note per bibliography resource
single_note

Before [18696d1](https://github.com/sul-dlss/exhibits/pull/715/commits/18696d1639063636d9a07b905db4f4b0479e8b27) we had several notes displayed inline, separated by with comma + and separators
inline_notes

## After
paragraph-separated-notes

### CODE REVIEW

**Comment by camillevilla** [comment]
## Option 1: `<ul>`
### multiple notes
unordered-list-notes

### single note
single-note

## Option 2: `<p>`
paragraph-separated-notes

**Comment by camillevilla** [comment]
Per discussion with @caaster and @blalbrit on 10/11, we'll be using paragraph separators for `general_notes_ssim` to generalize this field for other exhibits (notably, values of /mods/note without a displayLabel #543)

**Comment by ggeisler** [comment]
@camillevilla I like option 2 for presenting multiple notes, but do we have to worry about a given note having embedded paragraphs? If so, it might make it difficult to distinguish two different notes from a note with multiple paragraphs.

If notes with multiple paragraphs is not something we'll see, then I like the `<p>` approach.

**Comment by ggeisler** [comment]
@caaster @blalbrit I saw after my comment above that you had made the decision to use `<p>` for the separator. See my comment about a potential concern (that might not be relevant) about using a `<p>`.

**Comment by blalbrit** [comment]
Saw your comment @ggeisler - I think it's a good concern. We were looking at #543 as a larger "general notes" use-case that needs to be supported. The multi-paragraph note may be a thing for the larger use-case - we would need to do a bit more analysis.

@caaster - do you have more info from the MODS note domain that might cause us to reconsider the <p> approach?

**Comment by caaster** [comment]
Yes, Gary's comment about multiple paragraphs in a single note is an excellent one. Right now I can think of at least one use case where I think that occurs, but there could be more. Let me ask Arcadia to look at this ticket.

**Comment by arcadiafalcone** [comment]
Yes, the multiparagraph note question comes up fairly often. I've been recommending the use of <br> rather than <p>, though I can't answer for legacy data.

If #544 results in additional note index fields, they should behave in the same way.

Also relevant: #697 would apply the paragraph separator to all fields.

**Comment by camillevilla** [comment]
Thanks for the context, @arcadiafalcone! I'll experiment with a few different fixtures (multi-paragraph notes with `<br />` and `<p>` tags) and post samples here.

File: spec/fixtures/bibliography/book.bib [hunk 1]
```diff
@@ -10,5 +10,5 @@ @book{http://zotero.org/groups/1051392/items/TXXUJDG2
 	keywords = {sg012cr8689},
 	annote = { KAW},
 	annote = {CCC MS 470 [p. 9, and referred to (without MS number) throughout]reproductions from CCC MS 470, ff. 125-146, in the appendix},
```
Comment (drh-stanford) [inline spec/fixtures/bibliography/book.bib]: You're removing the multiple `annote` keys in other fixtures. Do you want to do the same for this fixture, or do you have a spec in mind for handling this case?

File: lib/traject/bibtex_config.rb [hunk 2]
```diff
@@ -38,7 +38,7 @@
 to_field 'issue_ssm', extract_bibtex_field(:issue)
 to_field 'pages_ssm', extract_bibtex_field(:pages)
 to_field 'doi_ssim', extract_bibtex_field(:doi)
-to_field 'general_notes_ssim', extract_bibtex_field(:annote, split: '|', trim: true)
+to_field 'general_notes_ssim', extract_bibtex_field(:annote, split: '\\textbar', trim: true)
```
Comment (drh-stanford) [inline lib/traject/bibtex_config.rb]: If you want, you can make this change in the preprocessing code so do a `gsub('\\textbar', '|')` here -- https://github.com/sul-dlss/exhibits/blob/master/app/models/bibliography.rb#L44

Comment (camillevilla) [reply lib/traject/bibtex_config.rb]: Thanks for the tip! That worked perfectly.

File: spec/fixtures/bibliography/book.bib (hunk 1 above)
Comment (camillevilla) [reply spec/fixtures/bibliography/book.bib]: Nope! Just a typo 😅

**Comment by camillevilla** [comment]
I've done a quick fix to handle the possibility of both `<p>` tags and `<br />` tags in the notes. 

### Unescaped HTML with paragraph separators
unescaped_html

### Escaped HTML with paragraph separator
Note the gap caused by double paragraph tags
double-paragraph tag

### Fix
single-wrap

This implementation skips wrapping the content in a `<p>` tag if it there is already a `<p>` tag present in the note.

**Comment by ggeisler** [comment]
@camillevilla I'm not sure if your focus here was solely on dealing with spacing issues when a note has embedded `<p>` or `<br>` tag. If so, great.

But I'm unsure if there is more work planned to focus on the presentation of multiple notes? If there is no further work planned on the presentation, I'm concerned, based on your screenshot above, that this is not a complete solution. Let me know and I'll provide more details of my concern if necessary.

**Comment by camillevilla** [comment]
@ggeisler this work was just focused on the spacing issues. Looking back at your comment, I see that I missed your finer point about distinguishing two different notes from a note with multiple paragraphs 😥

I'll make another ticket to make sure multiple notes get handled with a `<ul>` tag.

**Comment by ggeisler** [comment]
@camillevilla Okay, great. I know a `ul` in the metadata list is not particularly elegant; we could experiment with styling to make it a bit more subtle, such as making the bullets a gray rather than the darker default color, which would deemphasize them while retaining their value as an item indicator.

**Pull request merged**